### PR TITLE
[READY][TESTED]Plasmeme Gulag

### LINKED ISSUE #42472: Labor Camp Teleporter + Plasmeme = Death
When a plamaman is teleported to the gulag they're given the usual clothes, aka not an environment suit. They quickly burn to death after teleportation.

### PR DESCRIPTION
**Pull request by SouDescolado**
## About The Pull Request

```
//visibly unequips I but it is NOT MOVED AND REMAINS IN SRC
//item MUST BE FORCEMOVE'D OR QDEL'D
/mob/proc/temporarilyRemoveItemFromInventory(obj/item/I, force = FALSE, idrop = TRUE)
return doUnEquip(I, force, null, TRUE, idrop)
```

If I read this correctly, when you call this it just unequip. Since it returns true to being able to get removed, but returns false because it is on the plasmeme list, the item never get forcemoved, just removed.
## Why It's Good For The Game

Plasmemes can now go to gulag

## Changelog
🆑
fix: Benevolent Nanotrasen makes gulag available for everyone! (Plasmamen retain their equipment and don't die.)
/🆑

Web-pr. Will test tonight, but will leave here for feedback
Edit: Fixes #42472

### CODE REVIEW

**Comment by SouDescolado** [comment]
It seems that lots of things use it like this, but maybe plasmeme is the only one that can actually be punished by this?

Like, the light code:
```
/turf/open/floor/light/attackby(obj/item/C, mob/user, params)
	if(..())
		return
	if(istype(C, /obj/item/light/bulb)) //only for light tiles
		if(state && user.temporarilyRemoveItemFromInventory(C))
			qdel(C)
			state = 0 //fixing it by bashing it with a light bulb, fun eh?
			update_icon()
			to_chat(user, "<span class='notice'>You replace the light bulb.</span>")
		else
			to_chat(user, "<span class='notice'>The light bulb seems fine, no need to replace it.</span>")
```

If I am reading this correctly (which I may not), if you replace a bulb, then try to replace it again, it will remove the bulb, tell you the bulb is fine, but then it is not qdel or forcemoved.

@ExcessiveUseOfCobblestone You understand code better than me, is my assumption right?

**Comment by ExcessiveUseOfCobblestone** [comment]
Yes, hence the comment `item MUST BE FORCEMOVE'D OR QDEL'D`

**Review by ExcessiveUseOfCobblestone — DISMISSED** [review]

**Comment by SouDescolado** [comment]
For this PR I will only mess with plasmeme, but will go around in code trying to find all this other cases like this and make a big PR to fix them

**Comment by SouDescolado** [comment]
It works. The hardest part was to spawn myself, other than that works perfectly, with plasmaman teleporting with all its internals and handcuffed.

I put an ID on it that disappeared. Where it sends in meta?

**Comment by SouDescolado** [comment]
Found it. It works perfectly

File: code/game/machinery/gulag_teleporter.dm [hunk 1]
```diff
@@ -30,7 +30,8 @@ The console is located at computer/gulag_teleporter.dm
 		/obj/item/clothing/head/helmet/space/plasmaman,
 		/obj/item/tank/internals,
 		/obj/item/clothing/mask/breath,
-		/obj/item/clothing/mask/gas))
+		/obj/item/clothing/mask/gas,
+		/obj/item/restraints/handcuffs))
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/machinery/gulag_teleporter.dm]: um 👀

Comment (SouDescolado) [reply code/game/machinery/gulag_teleporter.dm]: I mean, this was already the case, wasn't it?

Comment (SouDescolado) [reply code/game/machinery/gulag_teleporter.dm]: Line 140 of the old code

Comment (ExcessiveUseOfCobblestone) [reply code/game/machinery/gulag_teleporter.dm]: sorry I was enjoying my weekend.

Can I ask how you are reaching that conclusion? Old L140-142 states that the handcuff moves to the top of the gulag teleporter (so you get your cuffs back).

Comment (SouDescolado) [reply code/game/machinery/gulag_teleporter.dm]: ....oh

**Comment by SouDescolado** [comment]
@ShizCalev Is it ok? I mean, it looks like it failed

**Comment by ExcessiveUseOfCobblestone** [comment]
lol

**Pull request merged**